MapboxNavigation/NavigationViewController.swift

```diff
@@ -368,6 +394,19 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
         if routeProgress.currentLegProgress.alertUserLevel == .arrive {
             navigationDelegate?.navigationViewController?(self, didArriveAt: destination)
         }
+        
+        updateStyleIfNeeded()
+    }
+    
+    func updateStyleIfNeeded() {
+        if let styles = styles, styles != [styleForTimeOfDayAndBrightness] {
+            self.styles = [styleForTimeOfDayAndBrightness]
+            applyStyle()
```

Comment:
@frederoni I'm getting mixed results when setting the style after it's been set. Any ideas as to why it does not apply the new style reliably?